sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -91,6 +91,9 @@ case class AnalyzeTableCommand(tableName: String, noscan: Boolean = true) extend
       case logicalRel: LogicalRelation if logicalRel.catalogTable.isDefined =>
         updateTableStats(logicalRel.catalogTable.get, logicalRel.relation.sizeInBytes)
 
+      case o if !o.isInstanceOf[LeafNode] =>
```

Comment:
I think a better fix is: when do `loopupRelation`, we always pass in a table identifier with database, so that it won't search the temp views.